ddtrace/contrib/botocore/patch.py

```diff
@@ -467,10 +495,37 @@ def patched_api_call(original_func, instance, args, kwargs):
         span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.botocore.get_analytics_sample_rate())
 
         try:
-            result = original_func(*args, **kwargs)
-            _set_response_metadata_tags(span, result)
+            if endpoint_name == "sqs" and operation == "ReceiveMessage" and config._data_streams_enabled:
+                queue_url = params["QueueUrl"]
+                queue_name = queue_url[queue_url.rfind("/") + 1 :]
+
+                if "MessageAttributeNames" not in args[1]:
+                    args[1].update({"MessageAttributeNames": ["_datadog"]})
+                elif "_datadog" not in args[1]["MessageAttributeNames"]:
+                    args[1].update({"MessageAttributeNames": args[1]["MessageAttributeNames"] + ["_datadog"]})
+
+                result = original_func(*args, **kwargs)
+                _set_response_metadata_tags(span, result)
+
+                for message in result["Messages"]:
+                    try:
+                        pathway = json.loads(message["MessageAttributes"]["_datadog"]["StringValue"])[
+                            PROPAGATION_KEY_BASE_64
+                        ]
+
+                        ctx = pin.tracer.data_streams_processor.decode_pathway_b64(pathway)
+                        ctx.set_checkpoint(["direction:in", "topic:" + queue_name, "type:sqs"])
+
+                    except Exception:
```

Comment:
DO we want to log a warning/error here?

Also catching exception is a bit broad. Is there a specific error we are try to handle? This code could always fail silently.